File: src/Island.js
import React from "react";
import PropTypes from "prop-types";

const Island = ({ name, svg, showBorder }) => (
  <div className={`${name.toLowerCase()}${showBorder ? " border" : ""}`}>
    <img src={svg} alt={name} />
  </div>
);

Island.propTypes = {
  name: PropTypes.string.isRequired,
  showBorder: PropTypes.bool
};

Island.defaultProps = {
  showBorder: false
};

export default Island;

File: danhodges+slinky/src/main/scala/example/hawaii/Island.scala
package example.hawaii

import slinky.core._
import slinky.core.annotations.react
import slinky.web.html._

import scala.scalajs.js

@react class Island extends StatelessComponent {
  case class Props(name: String, svg: js.Object, showBorder: Boolean)

  def render = {
    div(className := props.name.toLowerCase.concat(if (props.showBorder) " border" else ""))(
      img(src := props.svg.asInstanceOf[String], alt := props.name)
    )
  }
}

